pkg/install/upgrade.go

```diff
@@ -175,3 +176,11 @@ func (c *UpgradeController) configmapDataChanged(oldCM, newCM corev1.ConfigMap,
 	}
 	return false
 }
+
+func (c* UpgradeController) clearCache() {
```

Comment:
When we clear cache here, should we also remove locally saved secrets and configmaps as well?

Reply:
I don't think we need to remove them. Having a mismatch between the cache and locally saved secrets should cause a reconcile, and https://github.com/o-farag/hypershift-addon-operator/blob/c891a92db4f12cb78a4d454f3bf77a058ca23854/pkg/install/hypershift.go#L552 will update the secrets if anything changes